code/game/objects/effects/plastic_explosive.dm

```diff
@@ -0,0 +1,47 @@
+/obj/effect/plastic_explosive
+	name = "plastic explosives"
+	desc = "Used to put holes in specific areas without too much extra hole."
+	gender = PLURAL
+	icon = 'icons/obj/assemblies.dmi'
+	icon_state = "plastic-explosive2"
+	anchored = TRUE
+	density = FALSE
+	var/obj/item/plastique/parent
+
+/obj/effect/plastic_explosive/Initialize(var/atom/owner_pos, var/atom/target, var/obj/item/plastique/c4)
+	. = ..()
+	parent = c4
+	parent.effect_overlay = src
+	parent.forceMove(src)
+	name = parent.name
+	desc = parent.desc
+	set_position(get_dir(src, target))
+
+/obj/effect/plastic_explosive/proc/set_position(var/dir)
```

Comment:
Can this whole thing be done with the DIR2PIXEL_X and DIR2PIXEL_Y macros?

Reply:
No, that doesn't handle diagonals